web/packages/teleport/src/Discover/Database/DeployService/AutoDeploy/SelectSecurityGroups.tsx

```diff
@@ -200,21 +200,21 @@ export const SelectSecurityGroups = ({
 };
 
 function withTips(
-  securityGroups: SecurityGroup[],
-  db: AwsRdsDatabase
+  securityGroups?: SecurityGroup[],
+  db?: AwsRdsDatabase
 ): SecurityGroupWithRecommendation[] {
-  if (!db || !securityGroups) {
-    // return early without trying to add tips if there's no selected db or
-    // security groups given.
-    return securityGroups;
+  if (!securityGroups) {
```

Comment:
did you mean `!db`? (and remove optional for securitygroups?)

Reply:
I explained why I don't do `!db` in an updated comment (so we can still recommend security groups that allow outbound internet access for the auto-discovery flow)

Reply:
oh but yes, the security groups don't need to be optional and don't need to be checked for undefined.